### ISSUE TITLE
Open Windows Start menu on monitor with the mouse not working

### ISSUE THREAD

**Issue by gh-log** [valinet/ExplorerPatcher#1633]
doesnt work with Win Key. Opens in min monitor.

was working yesterday. 

latest Win 11 x64, latest EP.

shame I cant copy paste versions easily from EP.

**Comment by pyrates999** [comment]
Please check that symbols have downloaded by checking your notifications: https://github.com/valinet/ExplorerPatcher/wiki/Symbols

Then restart file explorer.  You can also try rebooting.  I had to do that after installing the July 11/2023 update.

**Comment by ninaran** [comment]
> Please check that symbols have downloaded by checking your notifications: https://github.com/valinet/ExplorerPatcher/wiki/Symbols
> 
> Then restart file explorer. You can also try rebooting. I had to do that after installing the July 11/2023 update.

So everything does work for KB5028185 after all? I had the issue of the start menu again not opening on the secondary monitor (where the taskbar is), instead always opening on primary. 
Even with using "When using multiple displays, open Start on this monitor when invoked using the keyboard : Monitor #2".

After the Windows update before that broke the functionality I'm hesitant to try updating again.

**Comment by pyrates999** [comment]
> > Please check that symbols have downloaded by checking your notifications: https://github.com/valinet/ExplorerPatcher/wiki/Symbols
> > Then restart file explorer. You can also try rebooting. I had to do that after installing the July 11/2023 update.
> 
> So everything does work for KB5028185 after all? I had the issue of the start menu again not opening on the secondary monitor (where the taskbar is), instead always opening on primary. Even with using "When using multiple displays, open Start on this monitor when invoked using the keyboard : Monitor #2".
> 
> After the Windows update before that broke the functionality I'm hesitant to try updating again.

You can choose not to update, but then you won't get security bugs fixed.

**Comment by Stealrull** [comment]
Yeah same issue, have it set to open on monitor 3 which is were the taskbar is but it opens on the primary monitor, small bug thanks to windows update but hopefully can be fixed at somepoint please!

**Comment by melMass** [comment]
The latest windows update broke it, it's actually functional (type and press enter it does work), but the menu is'nt visible

**Comment by Stealrull** [comment]
We reported a bug with the program pyrates999, that's the whole point of this board called 'issues', it's not working how the creator intended it too, no ones complaining simply reporting a bug.

Why are you hanging around here complaining at people reporting a feature of the program he wrote that bugged with the latest windows update? Makes no sense, clearly the creator wants it to work properly as it's an option to select which monitor it opens on, your argument is pointless. Find something else to do mate.

(and ninaran is correct)

**Comment by pyrates999** [comment]
> Because then all video games open up on that one. The whole point of this software for me is to have the taskbar + start menu on the monitor that is not my primary one.
> 
> And obviously "just don't update" isn't really an option for Windows security updates...

It was your idea not mine to not update.  Look at the quote I replied to.



> We reported a bug with the program pyrates999, that's the whole point of this board called 'issues', it's not working how the creator intended it too, no ones complaining simply reporting a bug.
> 
> Why are you hanging around here complaining at people reporting a feature of the program he wrote that bugged with the latest windows update? Makes no sense, clearly the creator wants it to work properly as it's an option to select which monitor it opens on, your argument is pointless. Find something else to do mate.
> 
> (and ninaran is correct)

I suggested a work around while this bug gets fixed.

**Comment by ninaran** [comment]
> > Because then all video games open up on that one. The whole point of this software for me is to have the taskbar + start menu on the monitor that is not my primary one.
> > And obviously "just don't update" isn't really an option for Windows security updates...
> 
> It was your idea not mine to not update. Look at the quote I replied to.

Okay then I'll keep it brief:
Is the bug of the start menu not opening up on the monitor it's set up to open still an issue in the latest July update? 

Simple yes/no is enough for me. Thanks.

**Comment by pyrates999** [comment]
> > > Because then all video games open up on that one. The whole point of this software for me is to have the taskbar + start menu on the monitor that is not my primary one.
> > > And obviously "just don't update" isn't really an option for Windows security updates...
> > 
> > 
> > It was your idea not mine to not update. Look at the quote I replied to.
> 
> Okay then I'll keep it brief: Is the bug of the start menu not opening up on the monitor it's set up to open still an issue in the latest July update?
> 
> Simple yes/no is enough for me. Thanks.

You can try to install the July 11/2023 update and check it yourself as I don't run windows 11, I run windows 10.

Then try the following to be sure it's not caused by an existing setting in explorer patcher:

1. uninstall explorer patcher
2. delete the registry key: "Computer\HKEY_CURRENT_USER\Software\ExplorerPatcher"
3. reboot
4. install explorer patcher
5. verify you get a notification that symbols have been downloaded
6. restart windows explorer using either the task manager or in the properties window of explorer patcher
7. reboot

Then apply your settings again in explorer patcher to see if it fixes it.

**Comment by ninaran** [comment]
Thank you.

I followed your steps as good as possible (except I installed the new July update in between 3 and 4) and can confirm that the bug is still present:
The Start menu ignores the setting to open on a monitor that is not the primary one in KB5028185. 

ExplorerPatcher Version 22621.1555.55.1
Windows 11 Version 22H2 (OS Build 22621.1992)

**Comment by brian6932** [comment]
Can reproduce

**Comment by pyrates999** [comment]
Check what happens when using the console in explorer patcher.  Open up the properties application for explorer patcher, go to advanced, and click on enable console.

Then try to reproduce this and show us what is in the console.

**Comment by ninaran** [comment]
```
funchook create 1
Running on Windows 11, OS Build 10.0.22621.1992.
Loaded symbols
Setup ntdll functions done
Setup user32 functions done
Patched taskbar transparency in newer OS builds
Setup explorer functions done
Setup uxtheme functions done
Setup twinui.pcshell functions done
Setup combase functions done
Setup twinui functions done
Setup stobject functions done
Setup bthprops functions done
Setup pnidui functions done
Setup shell32 functions done
Setup explorerframe functions done
Setup windows.storage functions done
[IME] Context menu patch status: 1
Setup inputswitch functions done
Setup windowsudk.shellcommon functions done
Setup peopleband functions done
Installed hooks.
Play startup sound thread...
Signal shell ready...
Started "Signal shell ready" thread.
Open Start on monitor thread
Started "Open Start on current monitor" thread.
EP Service Window thread
Initialized taskbar centering module.
Progman: 328418
Progman hook: 722679
[Positioning] Added settings for monitor 0000000000010001 : 0
[Positioning] Added settings for monitor 0000000000010003 : 0
 Signal shell ready.
Setup sndvolsso functions done
Ended "Signal shell ready" thread.
0 0
Started "Play startup sound" thread.
Ended "Play startup sound" thread.
[Updates] Starting daemon.
[Updates] Configured update policy on this system: "Check for updates but let me choose whether to download and install them".
[Updates] Path to module: C:\WINDOWS\dxgi.dll
[Updates] Checking against hash "22622.1555.55.1.5758b678431cded2"
[Updates] Update URL: https://github.com/valinet/ExplorerPatcher/releases/latest/download/ep_setup.exe
[Updates] Hash of remote file is "22622.1555.55.1.5758b678431cded2" (valid).
[Updates] Local version obtained from hash is 22622.1555.55.1.
[Updates] No updates are available.
```

This is the console output when I restart file explorer. 
I'm seeing a `Started "Open Start on current monitor" thread.` entry even though it should be on #2 as seen in the properties.

![image](https://github.com/valinet/ExplorerPatcher/assets/20947541/7c88bc9c-e6c0-41e5-b503-9713f8555283)

**Comment by ninaran** [comment]
A workaround I've found is to unlock the taskbar, move it over to the primary screen, let go and then move it back over to the secondary one and the Start menu then loads on the correct one. It's not perfect but it'll be good enough until a proper fix is out.

**Comment by gh-log** [comment]
I have a taskbar in all monitors.

**Comment by gh-log** [comment]
> Can reproduce, not specific to `mouse` tho, the title of the issue should be amended, it also occurs when using the Windows button or activating the start menu in any way at all. After some time tho, it did seem to fix itself, not sure why it took many explorer restarts.

start menu should open on monitor where mouse is so title is ok. used to work before.

if I move the mouse to monitor 1 and press WIN, SM should open there.
if I move the mouse to monitor 2 and press WIN, SM should open there.

**Comment by Sivart-Williamsson** [comment]
Same issue for me. Windows 11 OS build 22621.2215 ExplorerPatcher version 22621.1992.56.2.

**Comment by pyrates999** [comment]
> Same issue for me. Windows 11 OS build 22621.2215 ExplorerPatcher version 22621.1992.56.2.

That is a preview update: https://support.microsoft.com/en-us/topic/august-22-2023-kb5029351-os-build-22621-2215-preview-9af25662-083a-43f5-b3a7-975fe25cc692

Uninstall it if you want to use explorer patcher and don't install preview updates in the future: https://github.com/valinet/ExplorerPatcher/wiki/Symbols

In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.

All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.

Preview updates are also not fully tested by Microsoft and thus could have bugs.

**Comment by gh-log** [comment]
I had problems with 22621.2134 so 2215 is not the issue. The issue is way older than that as you can see in this page.

"Preview updates are also not fully tested by Microsoft and thus could have bugs." 

Release versions can also have bugs.

**Comment by ninaran** [comment]
Yeah it's no longer an issue of Preview updates, the "When using multiple displays, open Start on this monitor when invoked using the keyboard:" option in general doesn't work any more, sadly.

**Comment by ninaran** [comment]
Any news regarding this issue? Sure, you can circumvent it by moving the taskbar to the primary monitor and back to the secondary, but having to do that every single time you restart your PC gets bothersome.

**Comment by MochiesCode** [comment]
Also came here to report this bug. It seems to open on the correct monitor very temporarily if you change the "position on screen" option, but once you click on the start menu button with the mouse rather than using the keyboard key it goes back to not working again.

**Comment by pyrates999** [comment]
> Also came here to report this bug. It seems to open on the correct monitor very temporarily if you change the "position on screen" option, but once you click on the start menu button with the mouse rather than using the keyboard key it goes back to not working again.

Please post windows version and build.  You can find this by running winver.
Please post explorer patcher version.

Please include what settings you changed in explorer patcher properties application.

**Comment by Piehthyte** [comment]
I'm having the same issue, but only when I switch 'Position on screen' to 'at screen edge'.
If I set position on screen to center everything works fine.

**Comment by pyrates999** [comment]
when you press the windows button on the keyboard, what screen does the start menu open on?

**Comment by ninaran** [comment]
Primary monitor, no matter what setting I have selected.

**Comment by Piehthyte** [comment]
I'd like to also note that if you change the position of your start button from left-aligned to centered, click the start button while the mouse is on another display, and then revert the start button position to the left, it will subsequently open on the monitor where it was last accessed while in centered mode.

**Comment by pyrates999** [comment]
> I'd like to also note that if you change the position of your start button from left-aligned to centered, click the start button while the mouse is on another display, and then revert the start button position to the left, it will subsequently open on the monitor where it was last accessed while in centered mode.

Please update to the latest release of EP if you haven't.

**Comment by Roy379** [comment]
There is a still same problem for me. I hope this will be fixed soon.

**Comment by Piehthyte** [comment]
> > I'd like to also note that if you change the position of your start button from left-aligned to centered, click the start button while the mouse is on another display, and then revert the start button position to the left, it will subsequently open on the monitor where it was last accessed while in centered mode.
> 
> Please update to the latest release of EP if you haven't.

I have, sadly no change.

**Comment by MqcCheeze** [comment]
Still has this issue

**Comment by Piehthyte** [comment]
> Still has this issue
Can double confirm

**Comment by ninaran** [comment]
I have given up on this being fixed and moved over to StartAllBack instead: https://www.startallback.com/

**Comment by Brian-at-Baebies** [comment]
Issue still present as of this posting.  From reading the thread, it sounds like Microsoft did something that broke it, and it's up to them to undo that something for this to work again.  How sad.  I very briefly tried StartAllBack myself, but I've gotten used to the Win10 start menu, and SAB seemed not to give me anything between Win7 and Win11 look-and-feel.  At least ExplorerPatcher's Win10 start menu still works, at least with 26100.1742 (24H2).

**Comment by pyrates999** [comment]
If all you want is the windows 7 style start menu, open shell is what you want.
